src/analysis/normed_space/dual.lean

```diff
@@ -132,23 +136,17 @@ open metric set normed_space
 `polar 𝕜 s` is the subset of `dual 𝕜 E` consisting of those functionals which
 evaluate to something of norm at most one at all points `z ∈ s`. -/
 def polar (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
-  {E : Type*} [normed_group E] [normed_space 𝕜 E] (s : set E) : set (dual 𝕜 E) :=
-{x' : dual 𝕜 E | ∀ z ∈ s, ∥ x' z ∥ ≤ 1}
+  {E : Type*} [normed_group E] [normed_space 𝕜 E] : set E → set (dual 𝕜 E) :=
+(dual_pairing 𝕜 E).flip.polar
 
 variables (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
 variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
 
-@[simp] lemma zero_mem_polar (s : set E) :
-  (0 : dual 𝕜 E) ∈ polar 𝕜 s :=
-λ _ _, by simp only [zero_le_one, continuous_linear_map.zero_apply, norm_zero]
-
-lemma polar_eq_Inter (s : set E) :
-  polar 𝕜 s = ⋂ z ∈ s, {x' : dual 𝕜 E | ∥x' z∥ ≤ 1} :=
-by simp only [polar, set_of_forall]
+lemma mem_polar_iff {x' : dual 𝕜 E} (s : set E) : x' ∈ polar 𝕜 s ↔ ∀ z ∈ s, ∥x' z∥ ≤ 1 := iff.rfl
 
 @[simp] lemma polar_univ : polar 𝕜 (univ : set E) = {(0 : dual 𝕜 E)} :=
 begin
```

Comment:
Shouldn't this following directly from `linear_map.polar_univ`?

Reply:
mathematically yes, but there is some API missing namely `continuous_linear_map.coe_lm_injective`. I can PR that first.

Reply:
actually, this was not true, I just needed to do some unfolding to find the correct lemma